Backport of LHEGenericMassFilter to 106X

JanFSchulte opened the pull request:
This PR backports the LHE filter designed to cut on the invariant mass of a number of particles introduced in #35938. Backport to 10_6_X is necessary since some high dilepton mass ttbar and WW samples need to be produced in the UL campaigns.

JanFSchulte committed:
backport of LHEGenericMassFilter to 106X

cmsbuild commented:
A new Pull Request was created by  @JanFSchulte (Jan-Frederik Schulte) for CMSSW_10_6_X.

It involves the following packages:

- GeneratorInterface/GenFilters (**generators**)


@SiewYan, @mkirsano, @Saptaparna, @cmsbuild, @GurpreetSinghChahal, @agrohsje, @alberto-sanchez can you please review it and eventually sign? Thanks.
@alberto-sanchez, @mkirsano this is something you requested to watch as well.
@perrotta, @dpiparo, @qliphy you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

JanFSchulte committed:
code format and code checks

cmsbuild commented:
Pull request #36754 was updated. @SiewYan, @mkirsano, @Saptaparna, @cmsbuild, @GurpreetSinghChahal, @agrohsje, @alberto-sanchez can you please check and sign again.

agrohsje commented:
Hi @qliphy @perrotta , I removed myself from the l2-generators team but I still get mails about PR's. Does it take a while to update or should I remove my name somewhere else?

agrohsje commented:
please test

perrotta commented:
@agrohsje I still see your nick listed for "generators" in https://github.com/cms-sw/cms-bot/blob/master/categories.py

The following bot PR removes it: 
https://github.com/cms-sw/cms-bot/pull/1692
If it is actually what you wan't, please confirm and we'll merge it.

JanFSchulte commented:
I am not sure what the issue is here. In the tests I see only workflow 4.76 failing, which I can not check locally since the input files are not found at T2_CH_CERN. The code introduced here is not used in any of the tests, so I can't see how a failure would be related.

perrotta commented:
> I am not sure what the issue is here. In the tests I see only workflow 4.76 failing, which I can not check locally since the input files are not found at T2_CH_CERN. The code introduced here is not used in any of the tests, so I can't see how a failure would be related.

@JanFSchulte the crash in wf 4.76 is unrelated (see https://github.com/cms-sw/cmssw/issues/36771). Please disregard that workflow for now

SiewYan commented:
@perrotta , should i re-trigger the test and avoid 4.76 to complete the test?

JanFSchulte commented:
Could someone comment on the status of this PR? We would like to have if merged so we know we can move ahead with preparing the sample requests.

perrotta commented on GeneratorInterface/GenFilters/interface/LHEGenericMassFilter.h:
```diff
@@ -0,0 +1,50 @@
+class LHEGenericMassFilter : public edm::global::EDFilter<> {
+public:
+  explicit LHEGenericMassFilter(const edm::ParameterSet&);
+  ~LHEGenericMassFilter() override;
```
```suggestion
  ~LHEGenericMassFilter() override = default;
```

JanFSchulte replied:
Done

perrotta commented on GeneratorInterface/GenFilters/src/LHEGenericMassFilter.cc:
```diff
@@ -0,0 +1,62 @@
+      minMass_(iConfig.getParameter<double>("MinMass")),
+      maxMass_(iConfig.getParameter<double>("MaxMass")) {}
+
+LHEGenericMassFilter::~LHEGenericMassFilter() {
+  // do anything here that needs to be done at destruction time
+  // (e.g. close files, deallocate resources etc.)
+}
```
```suggestion
```

JanFSchulte replied:
Done

perrotta commented on GeneratorInterface/GenFilters/interface/LHEGenericMassFilter.h:
```diff
@@ -0,0 +1,50 @@
+
+private:
+  bool filter(edm::StreamID, edm::Event&, edm::EventSetup const&) const override;
+  void endJob() override;
```
Probably not needed

JanFSchulte replied:
Removed.

perrotta commented on GeneratorInterface/GenFilters/src/LHEGenericMassFilter.cc:
```diff
@@ -0,0 +1,62 @@
+
+  // event accept/reject logic
+  if (nFound == numRequired_) {
+    double Mass = std::sqrt(E * E - (Px * Px + Py * Py + Pz * Pz));
+    if (Mass > minMass_ && Mass < maxMass_) {
```
```suggestion
    double sqrdMass = (E * E - (Px * Px + Py * Py + Pz * Pz);
    if (sqrdMass > minMass_*minMass_ && Mass < maxMass_*maxMass_) {
```

JanFSchulte replied:
Done.

perrotta commented:
As in the backported PR, I would move it into the \plugin area, also combining the header and implementation file into a single .cc
By the way, why didn't you backport it as it was?

JanFSchulte committed:
merge files and move to plugins - also add BuildFile

cmsbuild commented:
Pull request #36754 was updated. @SiewYan, @mkirsano, @Saptaparna, @cmsbuild, @GurpreetSinghChahal, @alberto-sanchez can you please check and sign again.

JanFSchulte commented:
I have merged the files and moved them to plugins. I had split them up to conform to the structure of the other modules in this package, and because I had issues getting it to be included in the compilation when it was in the plugins folder. I realized now that was because there was no BuildFile.xml there, so now everything works.

JanFSchulte committed:
code checks and format

cmsbuild commented:
Pull request #36754 was updated. @SiewYan, @mkirsano, @Saptaparna, @cmsbuild, @GurpreetSinghChahal, @alberto-sanchez can you please check and sign again.

SiewYan commented:
please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-25e45e/22055/summary.html
**COMMIT**: cc765717f449d1fe1294901597dc29cc6346697b
**CMSSW**: CMSSW_10_6_X_2022-01-26-1100/slc7_amd64_gcc700
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/36754/22055/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 2 differences found in the comparisons
 - DQMHistoTests: Total files compared: 35
 - DQMHistoTests: Total histograms compared: 3215686
 - DQMHistoTests: Total failures: 2
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3215350
 - DQMHistoTests: Total skipped: 334
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 34 files compared)
 - Checked 143 log files, 29 edm output root files, 35 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_10_6_X_2022-01-26-1100+25e45e/48112/triggerResults): no differences found

SiewYan commented:
+1

its look good to me.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_10_6_X IBs (tests are also fine) and once validation in the development release cycle CMSSW_12_3_X is complete. This pull request will now be reviewed by the release team before it's merged. @perrotta, @dpiparo, @qliphy (and backports should be raised in the release meeting by the corresponding L2)

qliphy commented:
+1
